requre/online_replacing.py

```diff
@@ -443,6 +442,12 @@ def recording(
     cassette.data_miner.key_stategy_cls = storage_keys_strategy
     # ensure that directory structure exists already
     os.makedirs(os.path.dirname(cassette.storage_file), exist_ok=True)
+    # use default decorator for context manager if not given.
+    if (decorate is None and replace is None) or (
+        decorate is not None and replace is not None
```

Comment:
I am not sure about the second option, isn't this an error when `decorate is not None and replace is not None`?

Reply:
Right :-)
I've did not though that I've also covered this part.